isobus/src/isobus_virtual_terminal_server_managed_working_set.cpp

```diff
@@ -127,10 +128,46 @@ namespace isobus
 		return workingSetDeletionRequested;
 	}
 
+	void VirtualTerminalServerManagedWorkingSet::set_iop_size(std::uint32_t newIopSize)
+	{
+		const std::lock_guard<std::mutex> lock(managedWorkingSetMutex);
+		objectPoolTransferInProgress = true;
+		iopSize = newIopSize;
+	}
+
+	float VirtualTerminalServerManagedWorkingSet::iop_load_percentage() const
+	{
+		if (iopSize == 0)
+		{
+			return 0.0f;
+		}
+
+		auto totalTransferredSize = transferredIopSize;
+		if (totalTransferredSize < iopSize)
+		{
+			// if IOP transfer is not completed check if there is an ongoing IOP transfer to us
+			auto sessions = CANNetworkManager::CANNetwork.get_active_transport_protocol_sessions(0);
+			for (const auto &session : sessions)
+			{
+				if (session->get_source()->get_address() == get_control_function()->get_address())
```

Comment:
There is technically a bug here, though a minor one. You need to check the PGN of the session, and if it's not the ECU to VT one, continue instead of break, otherwise you will catch a destination specific request for ECU ID or something in here too, which is not what you want.